clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -5787,6 +5787,11 @@ void Clang::ConstructJob(Compilation &C, const JobAction &JA,
     CmdArgs.push_back("-treat-scalable-fixed-error-as-warning");
   }
 
+  // Enable local accessor to shared memory pass for SYCL.
+  if (isa<BackendJobAction>(JA) && IsSYCL) {
```

Comment:
@mdtoguchi, shouldn't the condition check `IsSYCLOffloadDevice` instead of `IsSYCL`?
`abi/user_mangling.cpp` and `regression/fsycl-save-temps.cpp` from check-sycl suite fails on my system.

Reply:
Ah, yes.  If this is to be only set for device, then this should be using `IsSYCLOffloadDevice`, as `IsSYCL` is for all compilations (host and device) when SYCL device offloading is enabled.

Reply:
I get following error:

clang (LLVM option parsing): Unknown command line argument '-sycl-enable-local-accessor'.  Try: 'clang (LLVM option parsing) --help'
clang (LLVM option parsing): Did you mean '--enable-local-reassign'?

Honestly, I don't really understand why the options is not visible for FE in host mode, but using `IsSYCLOffloadDevice` fixed the problem and I don't think running the pass enabled by `-sycl-enable-local-accessor` is needed in the host mode.
Another mystery is why this issue is not exposed by CI system. I suppose it some how related to the difference in cmake configuration - I don't build NVPTX and AMDGPU targets, which I suppose link the library with "unknown" option. 

We definitely need to do more investigation on this issue.

@jchlanda, FYI.

Reply:
Additional problem:  `-sycl-enable-local-accessor` is only being set when we do some kind of code generation step.  As the device compilation does not do this, `-sycl-enable-local-accessor` is never set for device.  It is only being emitted for host compilations as that goes through the assembling step.

Reply:
at a high level, `-sycl-enable-local-accessor` does only emit for the code generation step with the `nvptx64` target.  The steps are not combined for `nvptx64` allowing the option to only be emitted for device there.  Kind of a round-about way to restrict the option, but it can leak out to host if `-S` is used.

Reply:
IIRC, @jchlanda is away this week.
@AerialMantis, can someone else to take a look into this?

Reply:
That's right, I'll have someone else take a look.

Reply:
There is a PR up to fix this #5408
I believe the CI did not catch this as it builds for both cuda and hip then runs for each backend, correct me if I am wrong. If you build for CUDA or HIP then -sycl-enable-local-accessor is then usable.
I think that `-sycl-enable-local-accessor` is not available when not building for CUDA/HIP because the pass is initialised within the llvm NVPTX and AMDGPU backends.

Reply:
I'm almost glad that this bug surfaced @bader , @mdtoguchi . TBH, `-sycl-enable-local-accessor` is a workaround, that I never liked. The problem I had was that there wasn't a way to tell that a kernel was compiled from SYCL. Simply relying on the calling convention (`CallingConv::AMDGPU_KERNEL`) is not enough, as there are multiple paths that use it (OpenCL, OpenMP, SYCL). I was wondering if it would be better to follow NVIDIA here and use metadata nodes to denote kernels (https://github.com/intel/llvm/blob/HEAD/clang/lib/CodeGen/TargetInfo.cpp#L7242). This would work for all the passes that we only want to run on SYCL kernels (for instance we'd like to generalize https://github.com/intel/llvm/blob/sycl/llvm/lib/Target/NVPTX/SYCL/GlobalOffset.cpp would benefit from it).